cmd/autoconfigbrancher/main.go

```diff
@@ -0,0 +1,170 @@
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+
+	"github.com/openshift/ci-tools/cmd/autoconfigbrancher/helper"
+)
+
+const (
+	githubOrg   = "openshift"
+	githubRepo  = "release"
+	githubLogin = "openshift-bot"
+	githubTeam  = "openshift/openshift-team-developer-productivity-test-platform"
+)
+
+type options struct {
+	githubLogin    string
+	githubToken    string
+	gitName        string
+	gitEmail       string
+	targetDir      string
+	assign         string
+	currentRelease string
+	futureReleases flagutil.Strings
+}
+
+func parseOptions() options {
+	var o options
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flag.StringVar(&o.githubLogin, "github-login", githubLogin, "The GitHub username to use.")
+	flag.StringVar(&o.githubToken, "github-token", "", "The path to the GitHub token file.")
+	flag.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	flag.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	flag.StringVar(&o.targetDir, "target-dir", "", "The directory containing the target repo.")
+	flag.StringVar(&o.assign, "assign", githubTeam, "The github username or group name to assign the created pull request to.")
+	flag.StringVar(&o.currentRelease, "current-release", "", "Configurations targeting this release will get branched.")
+	flag.Var(&o.futureReleases, "future-release", "Configurations will get branched to target this release, provide one or more times.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login cannot be empty string")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	if o.targetDir == "" {
+		return fmt.Errorf("--target-dir is mandatory")
+	}
+	if o.assign == "" {
+		return fmt.Errorf("--assign is mandatory")
+	}
+	if o.currentRelease == "" {
+		return fmt.Errorf("--current-release is mandatory")
+	}
+	if len(o.futureReleases.Strings()) == 0 {
+		return fmt.Errorf("--future-release is mandatory")
+	}
+	return nil
+}
+
+func hasNewCommits() (bool, error) {
+	cmd := "git"
+	args := []string{"rev-list", "--count", "HEAD", "^before"}
+	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
+	stdout, _, err := helper.Call(cmd, args...)
+	if err != nil {
+		return false, err
+	}
+	i, err := strconv.Atoi(strings.TrimSuffix(string(stdout), "\n"))
+	if err != nil {
+		return false, err
+	}
+	return i > 0, nil
+}
+
+func run(cmd string, args ...string) {
+	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
+	if _, stderr, err := helper.Call(cmd, args...); err != nil {
```

Comment:
Why supress `stdout`?

Reply:
In this tool, we did not use `stdout` at all, except `git-rev-list`.
This function is an shortcut for log, call, fital in the `main` function.

Reply:
Right, but we likely want logs to actually show what the tools are doing in case we need to debug them at all.

Reply:
Not sure if I did what you want. So keep this comment open.

Reply:
Replace with the new command.
Resolving this one.